frontend/DataView/DataView.js

```diff
@@ -35,20 +35,51 @@ type DataViewProps = {
 class DataView extends React.Component {
   props: DataViewProps;
 
+  parseSparseArray(arr: Object) {
+    var acc: Array<any> = [];
+    var length = arr.length;
+    var numOfUndefsInRow = 0;
+    var pushUndefValsNumber = () => acc.push('undefined x ' + numOfUndefsInRow);
+
+    for (var i = 0; i < length; ++i) {
+      if (arr[i] === undefined) {
```

Comment:
It's a bit trickier that this.
Just because something is undefined, doesn't mean there is a hole in the array.

```
const a = [,,,]; // sparse
const b = [undefined, undefined, undefined]; // not sparse
```

I think the real check here is whether `arr.hasOwnProperty(i)`. But I might be wrong.